### PR TITLE
solver statistics is now a virtual function

### PR DESCRIPTION
**Pull request by hsitaram**
making solver statistics on rhsevals, steps etc.. into a virtual function that can now be implemented for cvode/arkode.

### CODE REVIEW

**Comment by marchdf** [comment]
Looks good to me! Other than maybe renaming to `print_final_stats` to stay within the "reactor" style maybe ;)

**Review by marchdf — APPROVED** [review]

File: Reactions/ReactorArkode.cpp [hunk 1]
```diff
@@ -110,8 +110,6 @@ ReactorArkode::init(int reactor_type, int /*ncells*/)
   } else {
     amrex::Print() << "ARK Step:" << std::endl;
   }
-  amrex::Print() << "  Setting tolerances rtol = " << relTol
```
Comment (marchdf) [inline Reactions/ReactorArkode.cpp:113]: did you mean to remove these? I am fine either way.

**Review by marchdf — APPROVED** [review]

File: Reactions/ReactorNull.H [hunk 2]
```diff
@@ -18,6 +18,8 @@ public:
 
   void close() override {}
 
+  void printFinalStats(void* mem) override {}
```
Comment (marchdf) [inline Reactions/ReactorNull.H]: since this is unused, can you `/*mem*/` ?

File: Reactions/ReactorRK64.H [hunk 3]
```diff
@@ -52,6 +52,8 @@ public:
 
   void close() override {}
 
+  void printFinalStats(void* mem) override {}
```
Comment (marchdf) [inline Reactions/ReactorRK64.H]: same for this one.

File: Reactions/ReactorNull.H (hunk 2 above)
Comment (hsitaram) [reply Reactions/ReactorNull.H]: done

File: Reactions/ReactorRK64.H (hunk 3 above)
Comment (hsitaram) [reply Reactions/ReactorRK64.H]: done

File: Reactions/ReactorArkode.cpp (hunk 1 above)
Comment (hsitaram) [reply Reactions/ReactorArkode.cpp:113]: yes, this was intended. The tolerances are printed out again through the set_sundials_tols function anyway when verbose is on.

**Review by esclapez — APPROVED** [review]
Looks good !

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**